poc/flp_pine.py

```diff
@@ -333,27 +326,27 @@ def eval_wr_checks(self,
 
         return (
             self.parallel_sum(mul_inputs),
-            wr_success_count - self.Field(self.num_wr_successes) * shares_inv
+            wr_success_count - self.field(self.num_wr_successes) * shares_inv
         )
 
-    def parallel_sum(self, mul_inputs):
+    def parallel_sum(self, mul_inputs: list[F]):
         """
         Split `mul_inputs` into chunks, call the gadget on each chunk, and
         return the sum of the results. If there is a partial leftover, then it
         is padded with zeros.
         """
-        s = self.Field(0)
+        s = self.field(0)
         while len(mul_inputs) >= 2*self.chunk_length:
             (chunk, mul_inputs) = front(2*self.chunk_length, mul_inputs)
-            s += self.GADGETS[0].eval(self.Field, chunk)
+            s += self.GADGETS[0].eval(self.field, chunk)
         if len(mul_inputs) > 0:
-            chunk = self.Field.zeros(2*self.chunk_length)
+            chunk = self.field.zeros(2*self.chunk_length)
             for i in range(len(mul_inputs)):
                 chunk[i] = mul_inputs[i]
-            s += self.GADGETS[0].eval(self.Field, chunk)
+            s += self.GADGETS[0].eval(self.field, chunk)
         return s
 
-    def encode_gradient_and_norm(self, measurement):
+    def encode(self, measurement: list[float]) -> list[F]:
```

Comment:
The new name suggests it implements `Valid.encode`. But actually they are not exactly the same. The latter requires both `encode_gradient_and_norm` and append results from `encode_wr_checks`, right?

Reply:
Done.